kernel/kernel_drivers.go

```diff
@@ -179,34 +202,129 @@ func RemoveKernelDriversTree(ksnapName string, rev snap.Revision) (err error) {
 	return os.RemoveAll(treeRoot)
 }
 
-// EnsureKernelDriversTree creates a tree in
-// /var/lib/snapd/kernel/<ksnapName>/<rev> with subfolders can be bind-mounted
-// on boot to /usr/lib/{modules,firmware}. This tree contains files from the
-// kernel snap mounted on kernelMount, as well as symlinks to it.
-// TODO this will be extended to consider kernel-modules components.
-func EnsureKernelDriversTree(ksnapName string, rev snap.Revision, kernelMount string) (err error) {
+// KernelModulesMntInfo contains information about a component and where it is
+// mounted.
+type KernelModulesMntInfo struct {
+	CompSideInfo *snap.ComponentSideInfo
+	MntDir       string
+}
+
+type KernelDriversTreeOptions struct {
+	// Set if we are building the tree for a kernel we are installing right now
+	KernelInstall bool
+}
+
+// EnsureKernelDriversTree creates a drivers tree that can include modules/fw
+// from kernel-modules components. opts.KernelInstall tells the function if
+// this is a kernel install (which might be installing components at the same
+// time) or an only components install.
+//
+// For kernel installs, a tree in /var/lib/snapd/kernel/<ksnapName>/<rev> with
+// subfolders can be bind-mounted on boot to /usr/lib/{modules,firmware} is
+// created. This tree contains files from the kernel snap mounted on
+// kernelMount, as well as symlinks to it. It gets used after a reboot.
+//
+// For components-only install, we want the components to be available without
+// rebooting. For this, we work on a temporal tree, and after finishing it we
```

Comment:
temporal feels a bit odd in this context
```suggestion
// rebooting. For this, we work on a temporary tree, and after finishing it we
```